cmd/kubeadm/app/util/kustomize/kustomize.go

```diff
@@ -243,7 +243,7 @@ func (km *Manager) Kustomize(data []byte) ([]byte, error) {
 
 	// Finally customize the target resource
 	var out bytes.Buffer
-	if err := kustomize.RunKustomizeBuild(&out, memFS, fakeDir); err != nil {
+	if err := kustomize.LegacyRunKustomizeBuild(&out, memFS, fakeDir); err != nil {
```

Comment:
cc @fabriziopandini

Reply:
I'm planning to ping the kubeadm folks over this, or just fix it myself in a distinct PR.

Reply:
@monopole may I ask why we are not switching to the new `RunKustomizeBuild`? Are we expected to make some change to make this happen?

Reply:
Switching to the new RunKustomizeBuild, or simply doing what that method does more directly, would be a good step to take.

This PR is only about changing kubectl, not kubeval.   A different PR should handle the kubeval change, as it will trigger a different cascade of vendoring changes.